reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -614,6 +618,97 @@ void sslExchangeRelativeGet() throws SSLException {
 		assertThat(responseString).isEqualTo("hello /foo");
 	}
 
+	@Test
+	void expectedConnectionPoolsEnabled() throws SSLException {
+		ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+		Logger spyLogger = Mockito.spy(log);
+		Loggers.useCustomLoggers(s -> spyLogger);
+
+		SslContext sslServer = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+				.build();
+
+
+		disposableServer =
+				createServer()
+						.secure(ssl -> ssl.sslContext(sslServer))
+						.handle((req, resp) -> resp.sendString(Flux.just("hello ", req.uri())))
+						.bindNow();
+
+		ConnectionProvider connectionProvider = ConnectionProvider.builder("expected-connection-pool").expectedConnectionPools(1).build();
+
+		StepVerifier.create(
+				Flux.range(1, 2)
+						.flatMap(i -> createClient(connectionProvider, disposableServer::address)
+								.secure(ssl -> ssl.sslContext(createClientSslContext()))
+								.get()
+								.uri("/foo")
+								.responseContent()
+								.aggregate()
+								.asString()))
+				.thenConsumeWhile(s -> true)
+				.verifyComplete();
+
+		Loggers.resetLoggerFactory();
```

Comment:
Please ensure you reset in finally block.

Reply:
sounds good. I'll go ahead and make all the suggested changes.